src/client/src/services/pricingService.ts

```diff
@@ -1,61 +1,79 @@
 import { Observable, Scheduler } from 'rxjs/Rx'
-import { ServiceBase } from '../system/service'
+import { ServiceClient } from '../system/service'
+import { ServiceConst } from '../types'
 import { logger, RetryPolicy } from '../system'
 import '../system/observableExtensions/retryPolicyExt'
 
 const log = logger.create('PricingService')
 const getPriceUpdatesOperationName = 'getPriceUpdates'
 
-export default class PricingService extends ServiceBase {
-  cachedObservablesBySymbol = {}
+interface Request {
+  symbol: string,
+}
 
-  private createSpotPriceStream(request) {
-    return Observable.create(
-      (o) => {
-        log.debug(`Subscribing to spot price stream for [${request.symbol}]`)
-        let lastPrice = null
-        return this.serviceClient
-          .createStreamOperation(getPriceUpdatesOperationName, request)
-          // we retry the price stream forever, if it errors (likely connection down) we pump a non tradable price
-          .retryWithPolicy(
-            RetryPolicy.indefiniteEvery2Seconds,
-            getPriceUpdatesOperationName,
-            Scheduler.async,
-            (err, willRetry) => {
-              if (willRetry && lastPrice !== null) {
-                // TODO: remove is stale price fully supported
-              }
-            },
-          )
-          .map(item => keysToLower(item))
-          .subscribe(
-            (price) => {
-              lastPrice = price
-              o.next(price)
-            },
-            (err) => { o.error(err) },
-            () => { o.complete() },
-          )
+const createSpotPriceStream = (serviceClient, request: Request) => {
+  const observable = Observable.create(clientObserver => {
+    log.debug(`Subscribing to spot price stream for [${request.symbol}]`)
+    let lastPrice = null
+    const innerObserver = {
+      next: price => {
```

Comment:
No need to use a fat arrow, there's no prototypical scope to retain.
```js
const innerObserver = {
  next(price) {
    lastPrice = price
    clientObserver.next(price)
  }
  …
  …
}
```

Reply:
Ok, I will do that. I think is more about a style preference. The compiled code in [production is exactly the same](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=MYewdgzgLgBAlmMBTATgeQEYVQN1TAXhgG8AoGGZADygAoAHFOYJAShPIpgBsBDaAApMWhGI2ZJOFYNzhIwUTNhR4UAOmp1xLVpwC-pA6VCRYCZOiy5UAJlFkKmgFxjhSQgD4OXHvyhCJUW1JHxk5BSVrdU0GN10KAz0gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=).

Reply:
Fair point! Thanks for including the example 😉